evalml/pipelines/components/transformers/imputers/knn_imputer.py

```diff
@@ -93,17 +92,19 @@ def transform(self, X, y=None):
         X_schema = X.ww.schema
 
         X_bool_nullable_cols = X_schema._filter_cols(include=["BooleanNullable"])
-        new_ltypes_for_bool_nullable_cols = {
-            col: "Boolean" for col in X_bool_nullable_cols
-        }
+        X_int_nullable_cols = X_schema._filter_cols(include=["IntegerNullable"])
```

Comment:
should we just use the downcasting function here?

Reply:
It seems that there's some issue with booleans being cast to floats when using `downcast_nullable_types`. I filed [this](https://github.com/alteryx/evalml/issues/3706) so we can get some consistency in behaviour using a single downcast function.